Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1373,6 +1373,18 @@ TextureCacheBase::AllocateTexture(const TCacheEntryConfig& config)
   return entry;
 }
 
+TextureCacheBase::TexPool::iterator
+TextureCacheBase::FindMatchingTextureFromPool(const TCacheEntryConfig& config)
+{
+  // Find a texture from the pool that does not have a frameCount of FRAMECOUNT_INVALID.
+  // This prevents a texture from being used twice in a single frame with different data,
+  // which potentially means that a driver has to maintain two copies of the texture anyway.
+  auto range = texture_pool.equal_range(config);
```

Comment:
Finding and allocating are two orthogonal things. Sure, they may be related in terms of use along side each other, but they're functionally separate things. Dropping the search algorithm code into a function that does allocation as its main operation isn't a great thing to do for organization and keeping functionally unrelated operations separate.